oif/helpers.py

```diff
@@ -93,6 +116,88 @@ def get_erc20_token_balances(token_address: str, holder_address: str, block_inte
 cg = CoinGeckoAPI()
 
 
+@cache
+def get_price_token_id(
+  coin_id: str,
+  t: Optional[int | str | datetime | date] = None,
+  currency: str = 'usd'
+) -> float:
+  """ Returns the price of coin with id :attr:`coin_id` at time :attr:`t`. If :attr:`t` is a string,
+  it is assumed to be in the required coingecko format (i.e.: DD-MM-YYYY). If it is an integer, then the
+  time is assumed to be a unix timestamp. If :attr:`t` is None, then the current price is returned.
+
+  Args:
+    coin_id (str): Coingecko coin id
+    t (int | str | datetime | date, optional): Time at which to get the price. Defaults to None.
+    currency (str, optional): Currency of the price value. Defaults to 'usd'.
+
+  Returns:
+    float: Coin price
+  """
+  if t is None:
+    return cg.get_price(coin_id, currency)[coin_id][currency]
+
+  match t:
+    case int() as timestamp:
+      date_ = date.fromtimestamp(timestamp)
+      date_str = f'{date_.day:02}-{date_.month:02}-{date_.year}'
+    
+    case date() | datetime() as date_:
+      date_str = f'{date_.day:02}-{date_.month:02}-{date_.year}'
+
+    case str() as date_:
+      date_str = date_
+
+  data = cg.get_coin_history_by_id(id=coin_id, date=date_str)
+  return data['market_data']['current_price'][currency]
+
+
+@cache
+def get_coin_id_of_address(address: str, network: str = 'ethereum') -> str:
+  """ Returns the coingecko ID of the ERC20 token with address :attr:`address` on
+  network :attr:`network`
+
+  Args:
+    address (str): The address of the ERC20
+    network (str, optional): The network on which the ERC20 is deployed. Defaults to 'ethereum'.
+
+  Raises:
+    Exception: The coingecko error
+
+  Returns:
+    str: The coingecko token ID
+  """
+  try:
+    data = cg.get_coin_info_from_contract_address_by_id(id=network, contract_address=address)
+    return data['id']
+  except KeyError:
+    raise Exception('token_of_address: {}'.format(data['error']))
```

Comment:
Theoretically, there could be a `KeyError` exception inside the Coingecko API call which this exception would accidentally catch. Though, likely it would happen, I try to minimize code inside exception calls.